packages/contracts/contracts/L2/predeploys/OVM_L1BlockNumber.sol

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.10;
+
+import { iOVM_L1BlockNumber } from "./iOVM_L1BlockNumber.sol";
+import { L1Block } from "../L1Block.sol";
+
+contract OVM_L1BlockNumber is iOVM_L1BlockNumber {
+    address public l1BlockAddress;
+
+    constructor(address _l1BlockAddress) {
```

Comment:
You should get rid of the constructor because that will require execution at deploy time, we want to be able to easily set the bytecode into the state when building the genesis file. Replace the constructor with an `initialize` function that `require`s the `l1BlockAddress == address(0)` then sets it so that it can be only called once